Funcky/Sequence/CycleRange.cs

```diff
@@ -14,6 +16,19 @@ public static partial class Sequence
         /// <returns>Returns an infinite IEnumerable repeating the same sequence of elements.</returns>
         public static IEnumerable<TItem> CycleRange<TItem>(IEnumerable<TItem> sequence)
             where TItem : notnull
+        {
+            var list = sequence.ToList();
+
+            if (list.None())
+            {
+                throw new ArgumentException("An empty sequence cannot be cycled", nameof(sequence));
```

Comment:
What's the rationale behind this decision?

Reply:
You cannot cycle an empty sequence, the current implementation simply hangs while trying to produce the first value.

We expect Cycle and CycleRange to produce infinite sequences, the alternative to return am empty sequence seems like misbehaviour.

Reply:
Ah yes, that's right :)